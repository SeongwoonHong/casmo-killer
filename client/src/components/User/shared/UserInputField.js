import React from 'react';
import PropTypes from 'prop-types';

import { trim } from 'sharedUtils/inputValidators';

const messages = (type) => {

  switch (type) {

    case 'email':
      return 'This email is linked to your account';

    case 'password':
      return 'Password must be between 6 and 20 characters.';

    case 'displayName':
      return 'Display name must be between 4 and 20 characters with no space.';

    default:
      return '';
  }

};

const placeholder = (type) => {

  switch (type) {

    case 'email':
      return 'Email Address';

    case 'password':
      return 'Password';

    case 'displayName':
      return 'Display Name';

    default:
      return '';
  }

};

const InputField = (props) => {
  if (props.isVisible) {
    return (
      <div className="user-form-fields">
        <label htmlFor={ props.name }>{ props.name }</label>
        <input
          type={ props.type }
          id={ props.name }
          name={ props.name }
          placeholder={ props.placeholder || placeholder(props.name) }
          onChange={ (e) => {
            props.onChange({
              name: e.target.name,
              value: trim(e.target.value)
            });
          } }
          value={ props.value }
          disabled={ props.disabled } />
        <p>{ props.message || messages(props.name) }</p>
      </div>
    );
  }
  return null;
};

InputField.propTypes = {
  isVisible: PropTypes.bool,
  name: PropTypes.string.isRequired,
  type: PropTypes.string,
  placeholder: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  message: PropTypes.string
};

InputField.defaultProps = {
  isVisible: true,
  type: 'text',
  placeholder: '',
  disabled: false,
  message: ''
};

export default InputField;
