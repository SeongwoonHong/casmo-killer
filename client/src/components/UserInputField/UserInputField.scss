.User__form__fields {
    border-bottom: 1px solid $main-menu-border-color;
    padding: $user-box-padding;
    &__label {
        color: palette(Grey, 800);
        display: block;
        font-size: 13px;
        font-weight: 600;
        text-transform: capitalize;
    }
    &__input {
        border: none;
        box-shadow:
            inset 0 0 0 1px $main-menu-border-color,
            0 0 0 transparent;
        font-size: 12px;
        margin: $user-box-padding 0 0;
        padding: 5px 7px;
        width: 100%;
        &:disabled {
            cursor: not-allowed;
        }
    }
    &__message {
        color: palette(Grey, 600);
        font-size: 12px;
        margin: 5px 0 0;
        text-align: right;
    }
    > .avatar-preview {
        @include flex-box(flex-start, flex-end);
        margin-top: $user-box-padding;
        > .avatar-wrapper {
            @include flex-box(center);
            @include rectanglify(100px);
            background-color: palette(Pink, 50);
            border: 1px solid palette(Pink, 100);
            border-radius: 50%;
            color: palette(Pink, 100);
            > .avatar-img {
                @include rectanglify(100px);
                border-radius: 50%;
            }
        }
        > input[type="file"] {
            position: absolute;
            z-index: -88;
            & + label {
                margin-left: $user-box-padding;
                text-align: center;
                > .User__form__btn {
                    color: palette(Grey, 50);
                    margin-bottom: $user-box-padding - 10px;
                }
                > span {
                    color: palette(Grey, 600);
                }
            }
        }
    }
}
