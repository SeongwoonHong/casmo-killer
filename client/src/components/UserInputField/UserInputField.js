import React from 'react';
import PropTypes from 'prop-types';

import { trim } from '@sharedUtils/inputValidators';

import './UserInputField.scss';

const inputType = (name) => {

  switch (name) {

    case 'password':
      return 'password';

    case 'email':
      return 'email';

    default:
      return 'text';

  }

};

const placeholder = (type) => {

  switch (type) {

    case 'email':
      return 'Email Address';

    case 'password':
      return 'Password';

    case 'displayName':
      return 'Display Name';

    default:
      return '';
  }

};

const messages = (type) => {

  switch (type) {

    case 'email':
      return 'This email is linked to your account.';

    case 'password':
      return 'Password must be between 6 and 20 characters.';

    case 'displayName':
      return 'Display name must be between 4 and 20 characters with no space.';

    default:
      return '';
  }

};

const UserInputField = (props) => {
  if (props.isVisible) {
    return (
      <div className="User__form__fields">
        <label
          htmlFor={ props.name }
          className="User__form__fields__label">
          { props.title || placeholder(props.name) }
        </label>
        <input
          className="User__form__fields__input"
          type={ props.type || inputType(props.name) }
          id={ props.name }
          name={ props.name }
          placeholder={ props.placeholder || placeholder(props.name) }
          onChange={ (e) => {
            props.onChange({
              name: e.target.name,
              value: trim(e.target.value)
            });
          } }
          value={ props.value }
          disabled={ props.disabled } />
        <p className="User__form__fields__message">
          { props.message || messages(props.name) }
        </p>
        {
          props.children
            ? props.children
            : null
        }
      </div>
    );
  }
  return null;
};

UserInputField.propTypes = {
  isVisible: PropTypes.bool,
  title: PropTypes.string,
  name: PropTypes.string.isRequired,
  type: PropTypes.string,
  placeholder: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  message: PropTypes.string,
  children: PropTypes.any,
};

UserInputField.defaultProps = {
  isVisible: true,
  title: '',
  type: '',
  placeholder: '',
  disabled: false,
  message: '',
  children: null
};

export default UserInputField;
