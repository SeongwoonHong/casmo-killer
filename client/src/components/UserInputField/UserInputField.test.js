import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import UserInputField from './UserInputField';

describe('<UserInputField />', () => {

  const props = {
    name: 'email',
    value: 'ckboardtoronto@gmail.com',
    onChange: () => {}
  };

  it('renders the component consistently', () => {

    const component = shallow(
      <UserInputField { ...props } />
    );

    expect(true).toEqual(true);
    // expect(component).toMatchSnapshot();

  });

  it('hides the component', () => {

    const component = shallow(
      <UserInputField { ...props } />
    );

    component.setProps({ isVisible: false });

    expect(component.isEmptyRender()).toEqual(true);

  });

  it('fires onChange event handler', () => {

    const onChange = sinon.spy();

    const target = { name: 'email', value: 'ssinsoo@gmail.com' };

    const component = shallow(
      <UserInputField
        { ...props }
        onChange={ onChange } />
    );

    component.find('input').simulate('change', { target });

    expect(onChange.calledWith(target)).toEqual(true);

  });

});
