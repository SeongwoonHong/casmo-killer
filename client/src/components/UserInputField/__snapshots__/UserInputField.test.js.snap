// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInputField /> renders the component consistently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserInputField
    disabled={false}
    isVisible={true}
    message=""
    name="email"
    onChange={[Function]}
    placeholder=""
    title=""
    type=""
    value="ckboardtoronto@gmail.com"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          className="User__form__fields__label"
          htmlFor="email"
>
          Email Address
</label>,
        <input
          className="User__form__fields__input"
          disabled={false}
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Email Address"
          type="email"
          value="ckboardtoronto@gmail.com"
/>,
        <p
          className="User__form__fields__message"
>
          This email is linked to your account.
</p>,
        null,
      ],
      "className": "User__form__fields user-form-fields",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Email Address",
          "className": "User__form__fields__label",
          "htmlFor": "email",
        },
        "ref": null,
        "rendered": "Email Address",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "User__form__fields__input",
          "disabled": false,
          "id": "email",
          "name": "email",
          "onChange": [Function],
          "placeholder": "Email Address",
          "type": "email",
          "value": "ckboardtoronto@gmail.com",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This email is linked to your account.",
          "className": "User__form__fields__message",
        },
        "ref": null,
        "rendered": "This email is linked to your account.",
        "type": "p",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="User__form__fields__label"
            htmlFor="email"
>
            Email Address
</label>,
          <input
            className="User__form__fields__input"
            disabled={false}
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Email Address"
            type="email"
            value="ckboardtoronto@gmail.com"
/>,
          <p
            className="User__form__fields__message"
>
            This email is linked to your account.
</p>,
          null,
        ],
        "className": "User__form__fields user-form-fields",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Email Address",
            "className": "User__form__fields__label",
            "htmlFor": "email",
          },
          "ref": null,
          "rendered": "Email Address",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "User__form__fields__input",
            "disabled": false,
            "id": "email",
            "name": "email",
            "onChange": [Function],
            "placeholder": "Email Address",
            "type": "email",
            "value": "ckboardtoronto@gmail.com",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This email is linked to your account.",
            "className": "User__form__fields__message",
          },
          "ref": null,
          "rendered": "This email is linked to your account.",
          "type": "p",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
